testutils/utils.go

```diff
@@ -43,3 +44,15 @@ func ProbeReadiness() (int, error) {
 	}
 	return resp.StatusCode, nil
 }
+
+func WaitForResourceSpecToUpdate(k8sClient client.Client, ctx context.Context, obj runtime.Object, timeout time.Duration, interval time.Duration) {
+	accessor, err := meta.Accessor(obj)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+	key := client.ObjectKey{Namespace: accessor.GetNamespace(), Name: accessor.GetName()}
+	scGen := accessor.GetGeneration()
+	Eventually(func() bool {
```

Comment:
Please use `EventuallyWithOffset(1, ...` instead of `Eventually(...` so we will have proper test output